portal/src/graphql/portal/query/nftContractMetadataQuery.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  LazyQueryResult,
+  OperationVariables,
+  useLazyQuery,
+} from "@apollo/client";
+import { useCallback } from "react";
+import { client } from "../apollo";
+import {
+  NftContractMetadataQueryQuery,
+  NftContractMetadataQueryDocument,
+} from "./nftContractMetadataQuery.generated";
+
+interface NftContractMetadataQueryResult {
+  fetch: (
+    contractId: string
+  ) => Promise<
+    LazyQueryResult<NftContractMetadataQueryQuery, OperationVariables>
+  >;
+  loading: boolean;
+  error: unknown;
+}
+
+export function useNftContractMetadataLazyQuery(
+  appID: string
+): NftContractMetadataQueryResult {
+  const [fetch, { loading, error }] =
+    useLazyQuery<NftContractMetadataQueryQuery>(
```

Comment:
Since I would like to get the result in a promise so that we can check the metadata before actually adding it to the list within the submit action, any better way to do this?